bin/improver-time-lagged-ensembles

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2018 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Script to run time-lagged ensembles."""
+
+import iris
+
+from improver.argparser import ArgParser
+from improver.utilities.load import load_cube
+from improver.utilities.save import save_netcdf
+from improver.utilities.time_lagging import GenerateTimeLaggedEnsemble
+
+
+def main():
+    """Load in the arguments and ensure they are set correctly. Then run
+    the time-lagged ensembles on the input cubes."""
+    parser = ArgParser(
+        description='This combines the realizations from different forecast '
+                    'cycles into one cube. It does this by taking an input '
+                    'CubeList containing forecasts from different cycles and '
+                    'merges them into a single cube, removing any metadata '
+                    'that does not match.')
+    parser.add_argument('input_filenames', metavar='INPUT_FILENAMES',
+                        nargs="+", type=str,
+                        help='Paths to input NetCDF files for the time-lagged '
+                        'ensemble to combine the realizations.')
+    parser.add_argument('output_file', metavar='OUTPUT_FILE',
+                        help='The output file for the processed NetCDF.')
+    args = parser.parse_args()
+
```

Comment:
At the moment, if only a single input filepath is specified rather than multiple filepaths, a `UserWarning `is generated from `cube_manipulation.py` - `Only a single cube so no differences will be found`. I'm not sure if this considered sufficient or if a check should made for this in the CLI....

Reply:
I will ask about this, and if it is required I'll add it in to the CLI.

Reply:
I looked at the weighted-blending CLI to see what it does.  In that case the CLI raises a warning and updates the metadata on the cube, and doesn't call the blending plugin.  So in this case I think a consistent approach would be to raise a warning from the CLI, and save the input cube (unmodified) to the output file name.

Reply:
Ok I will make those changes. Thanks!

Reply:
I have made these changes now, and added a warning within the CLI. I haven't added a CLI test (as neither of the other CLIs which use warnings have).

Reply:
This sounds sensible.  There is no need for a CLI test for this.  However can I suggest two things:
- A more descriptive warning message (eg `Only one cube provided - time lagging will have no effect`)
- Don't call the CLI in this case, but simply use `save_netcdf(cubes[0], args.output_file)`.  Wrap the remainder of the CLI in an "else" block so that things aren't saved twice.  This bypasses unnecessary calls and ensures the warning won't be raised at the cube combiner level.

Reply:
I'm happy with the changes, note the suggestions made by cgsandford.

Reply:
I just noticed that the travis build is failing - looks like you've got some extra whitespace:

`/home/travis/build/metoppv/improver/bin/improver-time-lagged-ensembles:65:1: W293 blank line contains whitespace`

Reply:
@cgsandford  I have made the changes you suggested with the warning.

@markysparks The changes should've resolved the travis issue too.